src/NHSD.BuyingCatalogue.Identity.Api/Startup.cs

```diff
@@ -76,13 +77,15 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton(cookieExpiration);
             services.AddSingleton(disabledErrorMessage);
             services.AddSingleton(registrationSettings);
+            services.AddSingleton<IScopeRepository>(new ScopeRepository(apiResources, identityResources));
```

Comment:
It might be a slight tidy to the code (slight as it is probably adding more code though) & help in the future if more dependencies get added to the `ScopeRepository`, but the suggestion here would be to register the `ApiResourceSettingCollection` & the `IdentityResourceSettingCollection` with the service collection & use dependency injection to create the `ScopeRepository`.

Reply:
I did actually consider this approach – in fact, it was my initial approach but I decided that (it was more verbose as you've commented) and that we wouldn't really gain anything from it at this point in time. As you say it would definitely be worth tweaking if we add more dependencies to the `ScopeRepository` but possible not right now...

What do you think? I'm happy with either approach.

Reply:
Lets stick with what we have currently. Just wanted to bring it up as a suggestion. Like you said we can revisit if more dependencies are required in the future 👍